app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -57,14 +57,14 @@ class BrowserRobot {
         val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
         sessionLoadedIdlingResource = SessionLoadedIdlingResource()
 
-        mDevice.waitNotNull(
-                Until.findObject(By.res("$packageName:id/mozac_browser_toolbar_url_view")),
-                waitingTime
-        )
-
         runWithIdleRes(sessionLoadedIdlingResource) {
-            onView(withId(R.id.mozac_browser_toolbar_url_view))
-                .check(matches(withText(containsString(url.replace("http://", "")))))
+            assertTrue(
+                mDevice.findObject(
+                    UiSelector()
+                        .resourceId("$packageName:id/mozac_browser_toolbar_url_view")
+                        .textContains(url.replace("http://", ""))
+                ).waitForExists(waitingTime)
+            )
```

Comment:
Simplified the URL verification and added the waitForExists method which proved more solid so far.